### PR TITLE
ChannelInitializeListenerHolder

### PR DESCRIPTION
**Pull request by Dysaido**
# Description

Add ChannelListener from paper-master, for better working of the viaversion

Fixes/Adds # (issue)

You can add channellister without reflection. 

# How has this been tested?

I have tested on own pc with viaversion, vulcan anticheat(packetevents) and I have added debug message to viaversion's injector. The console don't show error and these plugins work well.

### CODE REVIEW

File: NachoSpigot-API/pom.xml [hunk 1]
```diff
@@ -88,6 +88,36 @@
             <artifactId>annotations</artifactId>
             <version>23.0.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.checkerframework</groupId>
+            <artifactId>checker-qual</artifactId>
+            <version>3.21.0</version>
+        </dependency>
+        <!-- Kyori api -->
```
Comment (Lucaskyy) [inline NachoSpigot-API/pom.xml:96]: Why import the entire Kyori Adventure API if you're only going to use the `Key` class of it? Just download and import the class itself, don't import the entire API.

**Review by Lucaskyy — CHANGES_REQUESTED** [review]

File: NachoSpigot-API/pom.xml (hunk 1 above)
Comment (Dysaido) [reply NachoSpigot-API/pom.xml:96]: It was changed.

**Review by Elierrr — APPROVED** [review]

**Review by Lucaskyy — APPROVED** [review]

**Comment by Lucaskyy** [comment]
Thank you for this PR! :)

**Comment by HeathLoganCampbell** [comment]
Wow, @Dysaido great addition!

**Pull request merged**